Unable to build with Visual Studio 2019 version 16.2.0

# Environment
Windows build number: Version 10.0.18362.267

# Steps to reproduce
Update to the latest Visual Studio 2019 (16.2.0)
Build Terminal (you will get a compiler error)

(I had no issues building with 16.1.0)

# Expected behavior
Terminal should build without any errors

# Actual behavior
Here is the error:

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): error MSB3272: There was a problem reading the implementation file "C:\Users\Ajay\microsoft\terminal\x64\Release\TerminalAppLib\TerminalAppLib.lib". "Unable to read beyond the end of the stream."

<!-- What's actually happening? -->
Possibly related to https://github.com/microsoft/msbuild/issues/4547

--- a/src/cascadia/TerminalApp/FixVisualStudioBug.targets
+++ b/src/cascadia/TerminalApp/FixVisualStudioBug.targets
@@ -1,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--
+    This is a terrible, terrible rule. There exists a bug in Visual Studio 2019 16.2 and 16.3 previews
+    where ResolveAssemblyReferences will try and fail to parse a .lib when it produces a .winmd.
+    To fix that, we have to _temporarily_ replace the %(Implementation) on any winmd-producing
+    static library references with the empty string so as to make ResolveAssemblyReferences
+    not try to read it.
+
+    Upstream problem report:
+    https://developercommunity.visualstudio.com/content/problem/629524/static-library-reference-causes-there-was-a-proble.html
+  -->
+  <Target Name="_RemoveTerminalAppLibImplementationFromReference" BeforeTargets="ResolveAssemblyReferences">
+    <ItemGroup>
+      <_TerminalAppLibProjectReference Include="@(_ResolvedProjectReferencePaths)" Condition="'%(Filename)' == 'TerminalApp'" />
+      <_ResolvedProjectReferencePaths Remove="@(_TerminalAppLibProjectReference)" />
+      <_ResolvedProjectReferencePaths Include="@(_TerminalAppLibProjectReference)">
+        <Implementation />
+      </_ResolvedProjectReferencePaths>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_RestoreTerminalAppLibImplementationFromReference" AfterTargets="ResolveAssemblyReferences">
+    <ItemGroup>
+      <_ResolvedProjectReferencePaths Remove="@(_TerminalAppLibProjectReference)" />
+      <_ResolvedProjectReferencePaths Include="@(_TerminalAppLibProjectReference)" />
+    </ItemGroup>
+  </Target>
+  <!-- End "terrible, terrible rule" -->
+</Project>

--- a/src/cascadia/TerminalApp/TerminalApp.vcxproj
+++ b/src/cascadia/TerminalApp/TerminalApp.vcxproj
@@ -109,32 +109,8 @@
 
   <Import Project="$(OpenConsoleDir)src\cppwinrt.build.post.props" />
 
-  <!--
-    This is a terrible, terrible rule. There exists a bug in Visual Studio 2019 16.2 and 16.3 previews
-    where ResolveAssemblyReferences will try and fail to parse a .lib when it produces a .winmd.
-    To fix that, we have to _temporarily_ replace the %(Implementation) on any winmd-producing
-    static library references with the empty string so as to make ResolveAssemblyReferences
-    not try to read it.
-
-    Upstream problem report:
-    https://developercommunity.visualstudio.com/content/problem/629524/static-library-reference-causes-there-was-a-proble.html
-  -->
-  <Target Name="_RemoveTerminalAppLibImplementationFromReference" BeforeTargets="ResolveAssemblyReferences">
-    <ItemGroup>
-      <_TerminalAppLibProjectReference Include="@(_ResolvedProjectReferencePaths)" Condition="'%(Filename)' == 'TerminalApp'" />
-      <_ResolvedProjectReferencePaths Remove="@(_TerminalAppLibProjectReference)" />
-      <_ResolvedProjectReferencePaths Include="@(_TerminalAppLibProjectReference)">
-        <Implementation />
-      </_ResolvedProjectReferencePaths>
-    </ItemGroup>
-  </Target>
-
-  <Target Name="_RestoreTerminalAppLibImplementationFromReference" AfterTargets="ResolveAssemblyReferences">
-    <ItemGroup>
-      <_ResolvedProjectReferencePaths Remove="@(_TerminalAppLibProjectReference)" />
-      <_ResolvedProjectReferencePaths Include="@(_TerminalAppLibProjectReference)" />
-    </ItemGroup>
-  </Target>
-  <!-- End "terrible, terrible rule" -->
+  <!-- Import this set of targets that fixes a VS bug that manifests when using
+  the TerminalAppLib project -->
+  <Import Project="FixVisualStudioBug.targets" />
 
 </Project>

--- a/src/cascadia/ut_app/TerminalApp.UnitTests.vcxproj
+++ b/src/cascadia/ut_app/TerminalApp.UnitTests.vcxproj
@@ -122,4 +122,8 @@
     </PostBuildEvent>
   </ItemDefinitionGroup>
 
+  <!-- Import this set of targets that fixes a VS bug that manifests when using
+  the TerminalAppLib project -->
+  <Import Project="../TerminalApp/FixVisualStudioBug.targets" />
+
 </Project>

